hermes/apps/athena/modules/rbac-registration/src/database/select/cat_id.rs

```diff
@@ -64,55 +64,79 @@ use crate::{
 ///
 /// # Returns
 ///
-/// * `Ok(Vec<RbacChainInfo>)` – The registration chain associated with the given catalyst
-///   ID. If the vector is empty, no chain is found.
+/// * `Ok(Vec<RegistrationLocation>, RbacChainMetadata))` – The registration chain related
+///   data associated with the given catalyst ID. If the vector is empty, no chain is
+///   found.
 /// * `Err(anyhow::Error)` – If any error occurs.
 pub(crate) fn select_rbac_registration_chain_from_cat_id(
     persistent: &Sqlite,
     volatile: &Sqlite,
     cat_id: &str,
-) -> anyhow::Result<Vec<RbacChainInfo>> {
+) -> anyhow::Result<(Vec<RegistrationLocation>, RbacChainMetadata)> {
```

Comment:
Wouldn't it be more precise to return
```suggestion
) -> anyhow::Result<RbacChainData> {
```
where `RbacChainData` would be someshing like:
```rust
struct RbacChainData {
  locations: Vec<RegistrationLocation>,
  metadata: RbacChainMetadata
}
``` 
?
I'm not insisting on this, just trying to figure out🙃